api/filter_dimension_options.go

```diff
@@ -108,141 +179,174 @@ func (api *FilterAPI) getFilterBlueprintDimensionOption(w http.ResponseWriter, r
+	}
+
+	if filterBlueprint.State == models.SubmittedState {
+		return err
+	}
+
 	// FIXME - Once dataset API has an endpoint to check single option exists,
 	// refactor code below instead of creating an AddDimension object from the
 	// AddDimensionOption object (to be able to use checkNewFilterDimension method)
-	if err = api.checkNewFilterDimension(r.Context(), name, []string{option}, *filterBlueprint.Dataset); err != nil {
-		if err == filters.ErrVersionNotFound {
-			log.ErrorC("failed to select valid version", err, logData)
-			setErrorCode(w, err, statusUnprocessableEntity)
-			return
-		}
-
+	if err = api.checkNewFilterDimension(ctx, dimensionName, []string{option}, *filterBlueprint.Dataset); err != nil {
 		if incorrectDimensionOptions.MatchString(err.Error()) {
-			log.ErrorC("failed to select valid filter dimension options", err, logData)
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
+			return filters.NewBadRequestErr(err.Error())
 		}
 
 		if incorrectDimension.MatchString(err.Error()) {
-			log.ErrorC("failed to select valid filter dimension", err, logData)
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
+			return filters.NewBadRequestErr(err.Error())
 		}
 
-		log.ErrorC("failed to successfully check filter dimensions", err, logData)
-		setErrorCode(w, err)
-		return
+		return err
 	}
 
-	if err := api.dataStore.AddFilterDimensionOption(filterID, name, option); err != nil {
-		log.ErrorC("failed to add dimension option to filter blueprint", err, logData)
-		setErrorCode(w, err)
-		return
+	if err := api.dataStore.AddFilterDimensionOption(filterBlueprintID, dimensionName, option); err != nil {
+		return err
 	}
 
-	setJSONContentType(w)
-	w.WriteHeader(http.StatusCreated)
-
-	log.Info("created new dimension option for filter blueprint", logData)
+	return nil
 }
 
-func (api *FilterAPI) removeFilterBlueprintDimensionOption(w http.ResponseWriter, r *http.Request) {
+func (api *FilterAPI) removeFilterBlueprintDimensionOptionHandler(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
-	filterID := vars["filter_blueprint_id"]
-	name := vars["name"]
+	filterBlueprintID := vars["filter_blueprint_id"]
+	dimensionName := vars["name"]
 	option := vars["option"]
 
 	logData := log.Data{
-		"filter_blueprint_id": filterID,
-		"dimension":           name,
+		"filter_blueprint_id": filterBlueprintID,
+		"dimension":           dimensionName,
 		"option":              option,
 	}
 	log.Info("remove filter blueprint dimension option", logData)
 
-	filterBlueprint, err := api.getFilterBlueprint(r.Context(), filterID)
+	auditParams := common.Params{
+		"filter_blueprint_id": filterBlueprintID,
+		"dimension":           dimensionName,
+		"option":              option,
+	}
+	if auditErr := api.auditor.Record(r.Context(), removeOptionAction, actionAttempted, auditParams); auditErr != nil {
+		handleAuditingFailure(r.Context(), removeOptionAction, actionAttempted, w, auditErr, logData)
+		return
```

Comment:
Should we be setting an error status code in this case?

Reply:
`handleAuditingFailure` will set a 500 error if the audit fails. I have removed the return statement in this function below `logAuditFailure` (I wonder if this is what you referring to?)